astropy/coordinates/angles.py

```diff
@@ -584,7 +584,8 @@ def __setitem__(self, item, value):
         if isinstance(value, Longitude):
             raise TypeError("A Longitude angle cannot be assigned to a Latitude angle")
         # first check bounds
-        self._validate_angles(value)
+        if value is not np.ma.masked:
```

Comment:
I had hoped to avoid having non-masked classes to be aware of masks at all, but to avoid it here would mean creating a specific `MaskedLatitutde` class in `utils.masked`, which seems overkill. But the problem here is general, that classes may not be able to deal with `np.ma.masked`. I'll have a quick look to see if I cannot fix it more easily in `Masked` itself...

Reply:
That is an interesting question. In this case if there was going to be a `MaskedLatitude` class, wouldn't it belong in `angles.py` not in `masked.py`?  I can imagine a `MaskedLatitude` class that just overrides `_validate_angles` to do nothing if `value is np.ma.masked` otherwise calls the super method.

In general classes might need some small tweaks like this to become masked, and these might be classes outside of astropy. What's the pattern for doing that?

Reply:
So, my current idea is that the top-level `MaskedLatitude` class (which gets created by `Masked(Latitude)`) should really deal with this, i.e., define a `__setitem__`. I tried that locally and it works, but I need to think a little bit exactly how best to implement this... But if I don't get to it, I'm quite fine with just doing it here for now.

Reply:
In the spirit of "perfect is the enemy of good", I'd say to keep this as is for now. There will be other natural opportunities that come up to improve the implementation.

Reply:
Yes, agreed.